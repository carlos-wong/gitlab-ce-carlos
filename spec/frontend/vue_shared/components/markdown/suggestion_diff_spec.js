import { shallowMount } from '@vue/test-utils';
import SuggestionDiffComponent from '~/vue_shared/components/markdown/suggestion_diff.vue';
import SuggestionDiffHeader from '~/vue_shared/components/markdown/suggestion_diff_header.vue';
import SuggestionDiffRow from '~/vue_shared/components/markdown/suggestion_diff_row.vue';

const MOCK_DATA = {
  suggestion: {
    id: 1,
    diff_lines: [
      {
        can_receive_suggestion: false,
        line_code: null,
        meta_data: null,
        new_line: null,
        old_line: 5,
        rich_text: '-test',
        text: '-test',
        type: 'old',
      },
      {
        can_receive_suggestion: true,
        line_code: null,
        meta_data: null,
        new_line: 5,
        old_line: null,
        rich_text: '+new test',
        text: '+new test',
        type: 'new',
      },
      {
        can_receive_suggestion: true,
        line_code: null,
        meta_data: null,
        new_line: 5,
        old_line: null,
        rich_text: '+new test2',
        text: '+new test2',
        type: 'new',
      },
    ],
  },
  helpPagePath: 'path_to_docs',
};

describe('Suggestion Diff component', () => {
  let wrapper;

  const createComponent = () => {
    wrapper = shallowMount(SuggestionDiffComponent, {
      propsData: {
        ...MOCK_DATA,
      },
    });
  };

  beforeEach(() => {
    createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  it('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a correct amount of suggestion diff rows', () => {
    expect(wrapper.findAll(SuggestionDiffRow)).toHaveLength(3);
  });

  it('emits apply event on sugestion diff header apply', () => {
    wrapper.find(SuggestionDiffHeader).vm.$emit('apply', 'test-event');

    expect(wrapper.emitted('apply')).toBeDefined();
    expect(wrapper.emitted('apply')).toEqual([
      [
        {
          callback: 'test-event',
          suggestionId: 1,
        },
      ],
    ]);
  });
});
