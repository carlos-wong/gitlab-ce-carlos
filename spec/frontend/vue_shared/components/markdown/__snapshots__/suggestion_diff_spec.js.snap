// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestion Diff component matches snapshot 1`] = `
<div
  class="md-suggestion"
>
  <suggestion-diff-header-stub
    class="qa-suggestion-diff-header js-suggestion-diff-header"
    helppagepath="path_to_docs"
  />
   
  <table
    class="mb-3 md-suggestion-diff js-syntax-highlight code"
  >
    <tbody>
      <suggestion-diff-row-stub
        line="[object Object]"
      />
      <suggestion-diff-row-stub
        line="[object Object]"
      />
      <suggestion-diff-row-stub
        line="[object Object]"
      />
    </tbody>
  </table>
</div>
`;
